src/current/_plugins/sidebar_htmltest.rb

```diff
@@ -1,16 +1,36 @@
 require 'json'
 require 'liquid'
+require 'yaml'
 
 module SidebarHTMLTest
   class Generator < Jekyll::Generator
     def generate(site)
       @site = site
 
+      # Read htmltest configuration to get ignored directories
+      htmltest_config = YAML.load_file('.htmltest.yml') rescue {}
```

Comment:
this is duplicated from #21195 and #21416 - please merge those PRs and rebase, this should go away